src/atoms/Button/Button.test.js

```diff
@@ -0,0 +1,73 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Button from "./Button";
+
+describe("<Button />", () => {
+  const renderLikeSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        classFromProps="button--like"
+        buttonText="deixe seu like!"
+      />
+    );
+  };
+  const renderSubmitSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        buttonText="Adicionar"
+        classFromProps="button--submit"
+      />
+    );
+  };
+  it("renders a like button", () => {
+    renderLikeSetup();
+    const Button = screen.getByRole("button", { name: /like/i });
+    expect(Button).toBeInTheDocument();
+  });
+
+  it("renders a button with the correct submit class", () => {
+    renderSubmitSetup();
+    const submitButton = screen.getByRole("button", { name: /adicionar/i });
+    expect(submitButton).toHaveClass("button--submit");
+  });
+
+  it("renders a button with the correct like class", () => {
+    renderLikeSetup();
+    const Button = screen.getByRole("button", { name: /like/i });
+    expect(Button).toHaveClass("button--like");
+  });
+
+  it("calls the onClick function when clicked", () => {
+    const onClick = jest.fn();
+    renderLikeSetup(onClick);
+    const Button = screen.getAllByRole("button", { name: /like/i });
+    userEvent.hover(Button[0]);
+    userEvent.click(Button[0]);
```

Comment:
O seu teste renderiza apenas 1 button, acredito que não precise usar o `getAllByRole`, pode usar o `getByRole`:
```suggestion
    const button = screen.getByRole("button", { name: /deixe seu like!/i });
    userEvent.hover(button);
    userEvent.click(button);
```